### DISCUSSION TITLE
How to use undefined value?

### DISCUSSION THREAD

**Discussion by bhubavenski** [prisma/prisma#25218] [Q&A]
### Question

Hello, guys! I am having strange problem, why when formattedQuery is `undefined` it doesn't finds any posts, but if I refer directly `undefined` to `search` then it finds all posts? 
```js

  const formattedQuery = formatSearchQuery(searchInputParam);

  try {
      const posts = await PostRepo.findMany({
        where: {
          title: {
            search: formattedQuery,
          },
        },

```

For **example*** 
I can use
 
```
orderBy: {
 createdAt: undefined,
}
```

and this will work find, the posts will not be sorted by any mean

 but I can not use
 
```
const variable = undefined
orderBy: {
 createdAt: variable ,
}

```
This will not work! I am getting error: 

Error occurred while fetching posts: PrismaClientValidationError: 
Invalid `prisma.post.findMany()` invocation:

```
{
  where: {
    title: {
      search: "undefined"
    }
  },
  include: {
    author: true,
    tags: true,
    comments: {
      include: {
        author: true
      }
    }
  },
  orderBy: {
    createdAt: "undefined"
  }
}
```

### How to reproduce (optional)

try whatever query you want and see how if you use undefined as value directly, everything works as expected, but if you use it as a variable, then it will error

### Expected behavior (optional)

I expected to find all posts when the variable is undefined

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @bockster6669 👋 

We have improved the way undefined works in Prisma in our latest [release](https://github.com/prisma/prisma/releases/tag/5.20.0). Can you please update to the latest version 5.20.0 to get the benefits.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
